tests/parser_tests.py

```diff
@@ -315,6 +315,12 @@ def test_parse_subsecond_rounding(self):
         self.assertEqual(self.parser.parse(string, format), self.expected)
         self.assertEqual(self.parser.parse_iso(string), self.expected)
 
+        # overflow (zero out the subseconds and increment the seconds)
```

Comment:
Let's add a comment to denote that this is a regression test for issue https://github.com/crsmithdev/arrow/issues/636 so it can be referenced in the future if the issue reappears.